proposals/0107-unsaferawpointer.md

```diff
@@ -672,8 +670,8 @@ pointer cast is explicit:
 ```swift
 // --- new version ---
 func testInitAorB() {
-  let p = UnsafeMutableRawPointer.allocate(bytes: sizeof(Int.self),
-                                           alignedTo: alignof(Int.self))
+  let p = UnsafeMutableRawPointer.allocate(bytes: MemoryLayout<Int>.stride,
+                                           alignedTo: MemoryLayout<Int>.alignment)
 
   initRawAorB(p, isA: true)
   printA(p.assumingMemoryBound(to: A.self))
```

Comment:
@atrick Calling `initRawAorB(p, isA: true)` and then `initRawAorB(p, isA: true)` in this function violates the stated preconditions for `initializeMemory(as:to:)`, since the second call initializes memory for `B` that is already initialized for `A`. Is this permitted because `A` is a trivial type and `A` and `B` are layout compatible, or is deinitialization required here?